### DISCUSSION TITLE
pip install apache-airflow-upgrade-check upgrades airflow to latest version

### DISCUSSION THREAD

**Discussion by winston01** [apache/airflow#26480] [Q&A]
I'm in the process of upgrading airflow running in docker containers (based on _bitnami/airflow_) from 1.10.15 to 2.x. When I run `pip install apache-airflow-upgrade-check` within the container, it installs the latest version of airflow:

```
pip install apache-airflow-upgrade-check
WARNING: The directory '/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.
Collecting apache-airflow-upgrade-check
  Downloading apache_airflow_upgrade_check-1.4.0-py2.py3-none-any.whl (58 kB)
     |████████████████████████████████| 58 kB 15.9 MB/s
Collecting packaging
  Downloading packaging-21.3-py3-none-any.whl (40 kB)
     |████████████████████████████████| 40 kB 53.6 MB/s
Collecting apache-airflow<3,>=1.10.14
  Downloading apache_airflow-2.4.0-py3-none-any.whl (6.5 MB)
...
```

I may be wrong, but wouldn't the whole purpose of installing this package be to run `airflow upgrade_check` on the _old_ (1.10.15, that is) version, to check what should be upgraded?

Also, when the installation is done, and I run `airflow upgrade_check`, instead of getting the output described here [https://airflow.apache.org/docs/apache-airflow/stable/upgrading-from-1-10/upgrade-check.html#running-the-airflow-upgrade-check-package](url), I get warnings like this:

```
/env/lib/python3.8/site-packages/airflow/configuration.py:549 DeprecationWarning: The hide_sensitive_variable_fields option in [admin] has been moved to the hide_sensitive_var_conn_fields option in [core] - the old setting has been used, but please update your config.
...
```

Clearly, these are warnings for the preparation to upgrade to upcoming versions, not like the ones described on the above link.

Am I right to suggest that something is not right? If so, can you suggest a solution?

**Comment by winston01** [comment]
OK, I found the cause (not the solution, though): apparently bitnami uses apt-get to install airflow (in a certain location), thus pip install can't find it in the places it looks up to find it, so it'll install the latest available version as a dependency.

**Reply by potiuk** [reply]
Run `pip install apache-airflow-upgrade-check apache-airflow==1.10.15`

**Reply by winston01** [reply]
Unfortunately this doesn't work (not surprisingly, TBH), it installs the packages in a separate location (e.g., /opt/bitnami/airflow/venv/bin/airflow vs /opt/bitnami/python/bin/airflow). Unless bitnami has the apache-airflow-upgrade-check in their own repository (https://downloads.bitnami.com/files/stacksmith), I don't see a way to do this using their docker image. I think I'll have to move to using Apache's own docker-compose.yml.

**Reply by potiuk** [reply]
Or rather Apache Airlfow Docker image + Apache Airlfow Chart. Highly recomended .
